modules/api/src/main/java/org/apache/ignite/lang/ErrorGroups.java

```diff
@@ -73,13 +104,36 @@ public static synchronized ErrorGroup registerGroup(String groupName, short grou
             }
         }
 
-        ErrorGroup newGroup = new ErrorGroup(grpName, groupCode);
+        if (REGISTERED_ERROR_PREFIXES.add(errorPrefix)) {
+            String errorPrefixes = String.join("|", REGISTERED_ERROR_PREFIXES);
+            String pattern = EXCEPTION_MESSAGE_STRING_PATTERN.replace(PLACEHOLDER, errorPrefixes);
+            EXCEPTION_MESSAGE_PATTERN = Pattern.compile(pattern, DOTALL);
```

Comment:
do we really need to recompile pattern every time? I would prefer to go with more simple pattern, and do post validation like `REGISTERED_ERROR_PREFIXES.contains(prefix)` if this really necessary

Reply:
What does mean `every time`? Just once per each a new error prefix. I don't see benefits for moving it from start node to the first use